cmd/docker/docker.go

```diff
@@ -28,16 +28,43 @@ import (
 	"go.opentelemetry.io/otel"
 )
 
+var errCtxUserTerminated = errors.New("user terminated the process")
+
 func main() {
-	err := dockerMain(context.Background())
-	if err != nil && !errdefs.IsCancelled(err) {
+	ctx := context.Background()
+	err := dockerMain(ctx)
+	if err != nil && !errdefs.IsCancelled(err) && !errors.Is(err, errCtxUserTerminated) {
```

Comment:
`errors.Is(err, errCtxUserTerminated)` will never be true.  [CancelCauseFunc](https://pkg.go.dev/context#CancelCauseFunc) sets the cause of the context error and not the error itself.

Reply:
`ctx` errors are propagated up from the stack as normal errors. L35 will contain the context error making the check for `errors.Is(err, errCtxUserTerminated)` true. 

But you are correct that in this PR it won't ever be true since we are incorrectly wrapping errors downstream. A fix for that is in this PR https://github.com/docker/cli/pull/5666.